packages/blitz-auth/src/server/adapters/next-auth/types.ts

```diff
@@ -0,0 +1,32 @@
+import type {Ctx, MiddlewareResponse} from "blitz"
+import type {IncomingMessage} from "http"
+import type {AuthOptions, Profile, User} from "next-auth"
+import {SessionContext} from "../../../index-server"
+
+export type BlitzNextAuthOptions = AuthOptions & {
+  successRedirectUrl: string
+  errorRedirectUrl: string
+  secureProxy?: boolean
+  callback: (
+    user: User,
+    account: any,
+    profile: Profile,
```

Comment:
We want `profile` type to extract the actual profile type from the given `providers`. Do you need help with that?

Reply:
Yeah, This is where I spend most time in. I tried both the approach, the common callback and the individual callback.
I get mostly stuck with the circular or implicit invocation type errors.
I think I could use some help/insights.

I will post my attempts in Discord and could continue this there